src/main/java/domain/Car.java

```diff
@@ -0,0 +1,51 @@
+package domain;
+
+import exception.NotValidNameException;
+
+public class Car {
+
+    private String name;
+    private int currentLocation;
+
+    public Car(String name) {
+        this.name = carNameLengthValidation(name);
+        this.currentLocation = 0;
+    }
+
+    public Car(String name, int currentLocation) {
+        this.name = name;
+        this.currentLocation = currentLocation;
+    }
+
+    public static String carNameLengthValidation(String name) {
+        if (name.length() > 5) {
+            throw new NotValidNameException("자동차의 이름은 5글자를 초과할 수 없습니다.");
+        }
+        return name;
+    }
+
+    public Car move(Car car, boolean isMove) {
+        int isGo = car.getCurrentLocation();
+        if (isMove) {
+            return Car.of(car.getCarName(), isGo + 1);
+        }
```

Comment:
isGo라는 변수명은 boolean의 느낌이 있네요ㅎㅎ;
변수명이 짓기 어렵다면 아래와 같이도 가능할거같습니다 :)
```suggestion
        if (isMove) {
            return Car.of(car.name, car.currentLocation + 1);
        }
```